src/debugger/Debugger.js

```diff
@@ -97,48 +97,68 @@ export class DebugServer {
       if (filePath === null) return;
       let lineNum = location.start.line;
       let colNum = location.start.column;
-
       // Check whether there is a breakpoint we need to stop on here
-      if (!this.proceedBreakpoint(filePath, lineNum, colNum)) return;
-
+      let breakpoint = this.findStoppableBreakpoint(filePath, lineNum, colNum);
+      if (breakpoint === null) return;
       // Tell the adapter that Prepack has stopped on this breakpoint
-      this.channel.writeOut(`breakpoint stopped ${lineNum}:${colNum}`);
+      this.channel.writeOut(
+        `${DebugMessage.BREAKPOINT} ${DebugMessage.BREAKPOINT_STOPPED} ${breakpoint.filePath} ${breakpoint.line}:${breakpoint.column}`
+      );
 
       // Wait for the adapter to tell us to run again
-      this.waitForRun(function(line) {
-        return line === "proceed";
+      this.waitForRun((message: string) => {
+        return message === DebugMessage.PREPACK_RUN;
       });
     }
   }
 
-  executeCommand(command: string) {
+  // Process a command from a debugger. Returns whether Prepack should unblocks
+  // if it is blocked
+  processDebuggerCommand(command: string, runCondition?: string => boolean) {
```

Comment:
I originally thought there can be other commands to tell Prepack to continue depending on different reasons that it stopped. So runCondition is meant to capture any logic needed to determine whether the correct "continue" command was given in each case. As of now, only PREPACK_RUN is a continue command so I will change to remove the runCondition.

Reply:
There will be other commands telling Prepack to continue(like step-into, step-over, step-out etc..)
But still, process all these commands as a real command inside the big switch instead of using a out-of-band callback. The caller of this API should not need to decide what condition debugger should continue.